lib/FrontendTool/ReferenceDependencies.cpp

```diff
@@ -123,153 +137,155 @@ swift::reversePathSortedFilenames(const ArrayRef<std::string> elts) {
-      break;
-
-    case DeclKind::Enum:
-    case DeclKind::Struct:
-    case DeclKind::Class:
-    case DeclKind::Protocol: {
-      auto *NTD = cast<NominalTypeDecl>(D);
-      if (!NTD->hasName())
-        break;
-      if (NTD->hasAccess() &&
-          NTD->getFormalAccess() <= AccessLevel::FilePrivate) {
-        break;
-      }
-      out << "- \"" << escape(NTD->getName()) << "\"\n";
-      extendedNominals[NTD] |= true;
-      findNominalsAndOperators(extendedNominals, memberOperatorDecls,
-                               NTD->getMembers());
-      break;
-    }
+static void writeTopLevelDecl(
+    const Decl *D, llvm::raw_fd_ostream &out,
+    llvm::MapVector<const NominalTypeDecl *, bool> &extendedNominals,
+    llvm::SmallVectorImpl<const FuncDecl *> &memberOperatorDecls,
+    llvm::SmallVectorImpl<const ExtensionDecl *> &extensionsWithJustMembers) {
+  switch (D->getKind()) {
+  case DeclKind::Module:
+    break;
+
+  case DeclKind::Import:
+    // FIXME: Handle re-exported decls.
+    break;
+
+  case DeclKind::Extension:
+    writeExtensionDecl(cast<ExtensionDecl>(D), out, extendedNominals,
+                       memberOperatorDecls, extensionsWithJustMembers);
+    break;
+
+  case DeclKind::InfixOperator:
+  case DeclKind::PrefixOperator:
+  case DeclKind::PostfixOperator:
+    out << "- \"" << escape(cast<OperatorDecl>(D)->getName()) << "\"\n";
+    break;
+
+  case DeclKind::PrecedenceGroup:
+    out << "- \"" << escape(cast<PrecedenceGroupDecl>(D)->getName()) << "\"\n";
+    break;
+
+  case DeclKind::Enum:
+  case DeclKind::Struct:
+  case DeclKind::Class:
+  case DeclKind::Protocol:
+    writeNominalTypeDecl(cast<NominalTypeDecl>(D), out, extendedNominals,
+                         memberOperatorDecls);
+    break;
+
+  case DeclKind::TypeAlias:
+  case DeclKind::Var:
+  case DeclKind::Func:
+  case DeclKind::Accessor:
+    writeValueDecl(cast<ValueDecl>(D), out);
+    break;
+
+  case DeclKind::PatternBinding:
+  case DeclKind::TopLevelCode:
+  case DeclKind::IfConfig:
+  case DeclKind::PoundDiagnostic:
+    // No action necessary.
+    break;
+
+  case DeclKind::EnumCase:
+  case DeclKind::GenericTypeParam:
+  case DeclKind::AssociatedType:
+  case DeclKind::Param:
+  case DeclKind::Subscript:
+  case DeclKind::Constructor:
+  case DeclKind::Destructor:
+  case DeclKind::EnumElement:
+  case DeclKind::MissingMember:
+    // These can occur in malformed ASTs.
+    break;
+  }
+}
 
-    case DeclKind::TypeAlias:
-    case DeclKind::Var:
-    case DeclKind::Func:
-    case DeclKind::Accessor: {
-      auto *VD = cast<ValueDecl>(D);
-      if (!VD->hasName())
-        break;
-      if (VD->hasAccess() &&
-          VD->getFormalAccess() <= AccessLevel::FilePrivate) {
-        break;
-      }
-      out << "- \"" << escape(VD->getBaseName()) << "\"\n";
-      break;
-    }
+/// Write out the top level providers and collect decls for later writing
```

Comment:
"Provider" isn't really the right term. The file is the provider for all of these entries. "provided top-level names", maybe?

Reply:
What do you think of changing the terminology from "provided"/"providers" and "dependencies" to "exports" and "imports"? Or maybe "definitions" and "dependencies"?
"Name" seems not exactly right.

Reply:
"Imports" already has a meaning (module imports), so I'd rather not reuse that. "Definitions" doesn't work so well with the other kinds of "provides" entries, specifically the "nominal" one. Then again, that doesn't work so well with "provides" either. (See docs/DependencyAnalysis.rst for some elaboration on the model.)

I am happy to rename this because I don't like the grammatical problems with "provides", but I do want something that works with all the kinds of dependency, at least a little bit.

Reply:
Check out what I've done (when I push). Example: `writeProvidedTopLevelNames`. The biggest change is from "external" (which I thought was imprecise--external to what?) to "intermodule".